buffer.go

```diff
@@ -0,0 +1,196 @@
+package erasurecode
+
+import (
+	"bytes"
+	"io"
+)
+
+type BufferMatrix struct {
+	b                []byte
+	hdrSize, bufSize int
+	len              int // len of input
+	k                int
+	curBlock         int
+	leftInBlock      int
+	finished         bool
+}
+
+// FragLen returns the size of a "fragment" aligned to a block size (data + header)
+func (b BufferMatrix) FragLen() int {
+	return b.SubGroups() * (b.bufSize + b.hdrSize)
+}
+
+// SubGroups returns the number of blocks inside a single fragment
+func (b BufferMatrix) SubGroups() int {
```

Comment:
This value could be computed once then reused, it should not change while we read the http request right ?

Reply:
Yes and no. Due to Reset(), it may change everytime we recycle the buffer